### ISSUE TITLE
Generate text description of image prompt

### ISSUE THREAD

**Issue by mijanur132** [lucidrains/transfusion-pytorch#27]
The model currently can take text as input prompt and generate related image as output but the opposite (prompt: image, output: description of the image) does not seem to work. 

        prime = torch.tensor(image)
        one_multimodal_sample = model.sample(prime,max_length = 10)

1st issue is the assert error saying the starting tokens/prompts needs to be text tokens:
  transfusion.py 1405:    “assert last_modality_sample.dtype in (torch.int, torch.long), 'prompt must be text tokens'”
I tried several work-around to bypass above. 
1st one is to put the image prompt and then add the self.sos_id like below:
  
        init_mod_seq = tensor([self.sos_id], device = device)
        modality_sample = [ *default(prompt, []), init_mod_seq]

 2nd one is to remove the assertion completely and turning is_decoding_text to False to force modality/Image sample generation.

        modality_sample = [ *default(prompt, [])]

Also tried with self.som_ids[0]:

        init_mod_seq = tensor([self.som_ids[0]], device = device)
        modality_sample = [ *default(prompt, []), init_mod_seq]

None of this works as expected. 

I think having image to text part implemented and provided with an example usage would be a great addition to the transfusion model.

**Comment by lucidrains** [comment]
@mijanur132 thanks for opening this issue

indeed modality prompts are not working yet

as for an example of captioning, you can test it out here https://github.com/lucidrains/transfusion-pytorch/blob/main/train_mnist.py#L24

will get to this after I return from the holidays mid next week

**Comment by lucidrains** [comment]
@mijanur132 hey, noticed you deleted your other message

i take it you see that it is functional

i'm going to eliminate this issue in the next hour or so

**Comment by mijanur132** [comment]
Hi,

Good morning. I deleted that message because there was an problem from my
end which generated the run time error which I fixed. Now that script and
unprompted image captioning worked.
But the original issue which is giving image as prompt to generate text is
still there. So i think the issue needs further attention.

Thank you.

Best Regards,
Dr. Mijanur Palash
HPC & AI Post-doctoral Associate
Oak Ridge National Lab, TN, USA
